src/app/home-component/home-component.component.ts

```diff
@@ -49,6 +49,7 @@ export class HomeComponentComponent implements OnInit {
 
   ngOnInit(): void {
     this.getCarList();
+    console.log(this.getCarList());
```

Comment:
Tu dostajesz undefined ponieważ ta funkcja nic nie zwraca. To jest observable, który obsługujesz w lini 63